unyt/unit_object.py

```diff
@@ -480,18 +480,16 @@ def __pow__(self, p):
 
     def __eq__(self, u):
         """Test unit equality."""
-        if not isinstance(u, Unit):
-            return False
         return (
-            math.isclose(self.base_value, u.base_value)
+            isinstance(u, Unit)
             and self.dimensions == u.dimensions
+            and math.isclose(self.base_value, u.base_value)
+            and math.isclose(self.base_offset, u.base_offset)
         )
 
     def __ne__(self, u):
         """Test unit inequality."""
-        if not isinstance(u, Unit):
-            return True
-        if not math.isclose(self.base_value, u.base_value):
+        if not self == u:
```

Comment:
I don't remember exactly why but I'm pretty sure there was a reason `__ne__` wasn't implemented in terms of `==`. Speed maybe? Avoids one extra function call and the indirection of going from `==` to `__eq__` in CPython.

If there isn't a big speed difference and you do end up doing it this way there's no need to keep the rest of the implementation of `__ne__`.